config/locales/en.yml

```
@@ -1,33 +1,22 @@
-# Files in the config/locales directory are used for internationalization
-# and are automatically loaded by Rails. If you want to use locales other
-# than English, add the necessary files in this directory.
-#
-# To use the locales, use `I18n.t`:
-#
-#     I18n.t "hello"
-#
-# In views, this is aliased to just `t`:
-#
-#     <%= t("hello") %>
-#
-# To use a different locale, set it with `I18n.locale`:
-#
-#     I18n.locale = :es
-#
-# This would use the information in config/locales/es.yml.
-#
-# The following keys must be escaped otherwise they will not be retrieved by
-# the default I18n backend:
-#
-# true, false, on, off, yes, no
-#
-# Instead, surround them with single quotes.
-#
-# en:
-#   "true": "foo"
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
 en:
-  hello: "Hello world"
+  books:
+    index:
+      new_book: New book
+      show_this_book: Show this book
+    new:
+      new_book: New book
+      back_to_books: Back to books
+    show:
+      edit_this_book: Edit this book
+      back_to_books: Back to books
+      destroy_this_book: Destroy this book
+    edit:
+      editing_book: Edit this book
+      show_this_book: Show this book
+      back_to_books: Back to books
+    create:
+      success: Book was successfully created.
```

```suggestion
      success: %{name} was successfully created.
```

のようにパラメータを使えば本以外の画面でも再利用しやすくなります。
（その場合は翻訳キーも再利用しやすい名前にしてください）

---

式展開を使えば再利用しやすくなるの具体的なイメージがつかなかったので、(ヒントでも良いので)教えていただきたいです。

私が考えたのは

たとえば `en.yml` に以下に式展開した書き方をします
```yaml
  flash_message:
    create: "%{name} was successfully created."
    update: "%{name} was successfully updated."
    destroy: "%{name} was successfully destroyed."
```

- 本の画面で利用する場合
`books_controller` にロケーションファイルの翻訳情報を取得するように書きます
```ruby
def create
(省略)
        (省略), notice: t('flash_message.create', name: @book.title) }
(省略)
end
```

この場合、本が登録されたときに、「{本のタイトル} was successfully created.」というメッセージが表示されます

- ユーザの画面で利用する場合
`users_controller` にロケーションファイルの翻訳情報を取得するように書きます。
```ruby
def create
(省略)
        (省略), notice: t('flash_message.create', name: @user.name) }
(省略)
end
```

この場合、ユーザが登録されたときに、「{ユーザ名} was successfully created.」というメッセージが表示され、ロケーションファイルを変更しなくても良いので、再利用しやすくなると考えたました

---

再利用のイメージはだいたい合ってますが、`t('flash_message.create', name: @book.title)`だと「吾輩は猫である was successfully created.」のようなメッセージになりますね。デフォルトだと"Book was successfully created."なので、同じ文言になるのが理想的です。